app/cluster/state/etcd.go

```diff
@@ -0,0 +1,279 @@
+package state
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/rudderlabs/rudder-server/app/cluster"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/utils/types/servermode"
+	"github.com/rudderlabs/rudder-server/utils/types/workspace"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
```

Comment:
```suggestion
	etcdClientv3 "go.etcd.io/etcd/client/v3"
```

Reply:
using `snakeCase` is not recommended for package names, I could use `etcdclientv3`, but it would be ugly/hard to read. 

How about just `etcd` ?

Reply:
Indeed.

> When renaming an imported package, the local name should follow the same guidelines as package names (lower case, no under_scores or mixedCaps).

Reference: https://go.dev/blog/package-names?utm_source=dlvr.it&utm_medium=twitter

`etcd` or `etcd3` both look good to me.

Reply:
it is used as clientV3 even on kubernetes repo so I would say we can just go ahead with what we have right now.